File: 2021/May/Day 10/three-consequtive-odds.py
# SOurce : https://leetcode.com/problems/three-consecutive-odds/submissions/
class Solution(object):
    def threeConsecutiveOdds(self, arr):
        """
        :type arr: List[int]
        :rtype: bool
        """
        count =0;
        for i in range(len(arr)-2):
            sub_array = arr[i:i+3]
            if(self.check_three_odds(sub_array)):
                return True
        return False

    def check_three_odds(self,arr):
        if(len(arr)<3):
            return False
        return arr[0]%2==1 and arr[1]%2==1 and arr[2]%2==1

File: 2022/June /Day 21/threeConsequtiveOdds.cpp
// Source : https://leetcode.com/problems/three-consecutive-odds/submissions/
class Solution {
public:
        bool isOdd(int n1,int n2,int n3)
        {
            return n1%2==1 && n2%2==1 && n3%2==1;
        }
    bool threeConsecutiveOdds(vector<int>& arr) {

        if(size(arr)<3){return false;}

        for(int i=0;i<size(arr)-2;i++)
        {
            if(isOdd(arr[i],arr[i+1],arr[i+2])){return true;}
        }
        return false;
    }
};
